replicatedhq/ship
# platform behavior for how ship renders defaults and values

What I Did
------------
Changed the behavior of how `default` and `value` config options are rendered. Ship config now behaves like platform config.

How I Did it
------------
- Modified how defaults are applied to values
- Modified how a config item is rendered based on the source of its value
- Added logic to handle this behavior in a headed install or headless install (it's slightly different)

How to verify it
------------
`default-values` integration tests

Description for the Changelog
------------
General behavior is as follows:
- a default value is never written to `state.json`
- a supplied value can, and will, override a default value and will be written to `state.json`

Edge-case behavior:
- a value can take on an empty value, e.g. `""`, only when it is supplied by `state.json`. this value will behave identically to how a regular value would behave (see above)
- if a supplied value is equivalent to its corresponding default value, e.g., `value: abc`, `default: abc`, the value will behave identically to how a regular value would behave (see above)
- if both default and value are not supplied, the value will behave identically to how a regular value would behave (see above)

Picture of a Boat (not required but encouraged)
------------
🛥 🛥
